Tools/peg_generator/pegen/c_generator.py

```diff
@@ -29,6 +29,13 @@
 EXTENSION_PREFIX = """\
 #include "pegen.h"
 
+#ifdef Py_DEBUG
+extern int Py_DebugFlag;
+#define D(x) if (!Py_DebugFlag); else x
```

Comment:
Cross-posting from our Hangouts chat:

I've got difficulties understanding this line. If you ever find the time, could you very briefly explain it?

Reply:
That comes from the old parser: https://github.com/python/cpython/blob/master/Parser/parser.c#L17-L22. WHat's happening basically is that there is a global symbol Py_DebugFlag that is 1 when -d is activated (the argument parsing sets it). If you compile python without debug build those lines are eliminated because D(x) expands to nothing, and if you compile Python in debug build then D(x) is expanded to x surrounded by a check for the Py_DebugFlag flag

The idea is that checking for the -d flag and the associated code has no cost on release mode

Reply:
Ooh, understood. Is then then equivalent to this?

```
#define D(x) if (Py_DebugFlag) x;
```

If so, is there a reason why the former is preferred?

Reply:
> If so, is there a reason why the former is preferred?

Not that I am aware of, I just ported that chunk from the old parser. Will change it to `#define D(x) if (Py_DebugFlag) x;` for clarity